internal/proto/reader.go

```diff
@@ -55,10 +55,13 @@ func (r *Reader) Reset(rd io.Reader) {
 }
 
 func (r *Reader) ReadLine() ([]byte, error) {
-	line, err := r.readLine()
-	if err != nil {
+	line, err := r.rd.ReadBytes('\n')
```

Comment:
Supposedly this allocates a new byte slice for each read line. If so, then it is a significant performance penalty for something fast like Redis client.

Reply:
Instead you probably need to modify `readLine` to properly handle `bufio.ErrBufferFull`.

Reply:
Thanks. I will look into it further.